Compiler/script2/cs_parser.h

```diff
@@ -552,35 +552,24 @@ class Parser
 
     static int GetWriteCommandForSize(int the_size);
 
-    // Handle the cases where a value is a whole array or dynarray or struct
-    void HandleStructOrArrayResult(EvaluationResult &eres);
-
     // If the result isn't in AX, move it there. Dereferences a pointer
     void EvaluationResultToAx(EvaluationResult &eres);
 
-    // We're in the parameter list of a function call, and we have less parameters than declared.
-    // Provide defaults for the missing values
-    void AccessData_FunctionCall_ProvideDefaults(int func_args_count, size_t supplied_args_count, Symbol func_symbol, bool func_is_import);
-
-    void AccessData_FunctionCall_PushParams(SrcList &params, size_t closed_paren_idx, size_t func_args_count, size_t supplied_args_count, Symbol funcSymbol, bool func_is_import);
-
-    // Count parameters, check that all the parameters are non-empty; find closing paren
-    void AccessData_FunctionCall_CountAndCheckParm(SrcList &parameters, Symbol name_of_func, size_t &index_of_close_paren, size_t &supplied_args_count);
+    // We are processing a function call. Emit the actual function call
+    void AccessData_FunctionCall_EmitCall(Symbol name_of_func, size_t params_count, bool func_is_import);
 
-    // We are processing a function call. General the actual function call
-    void AccessData_GenerateFunctionCall(Symbol name_of_func, size_t args_count, bool func_is_import);
-
-    // Generate the function call for the function that returns the number of elements
+    // Emit the function call for the function that returns the number of elements
     // of a dynarray.
     void AccessData_GenerateDynarrayLengthAttrib(EvaluationResult &eres);
```

Comment:
I did a change to how Array.Length is implemented not a long time ago: #2559

In short, this method no longer generates *the function call*, instead it dynamically generates and registers *attribute and getter entries*, and then lets parser proceed as usual.

Apparently I forgot to fix the comment here. It should be something like "Generate Length attribute for the parsed array type and add it to the symbol table", for example.

Reply:
Will enter the comment.